homeassistant/components/mqtt/mixins.py

```diff
@@ -1225,6 +1226,57 @@ def _prepare_subscribe_topics(self) -> None:
     async def _subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
 
+    @callback
+    def _attrs_have_changed(
+        self, attrs_snapshot: tuple[tuple[str, Any | UndefinedType], ...]
+    ) -> bool:
+        """Return True if attributes on entity changed or if update is forced."""
+        if self._attr_force_update:
+            return True
```

Comment:
This is not used yet, so I suggest we add this when we implement this on `sensor` and `binary_sensor` to ensure code coverage.

```suggestion
```

Reply:
Maybe do both `sensor` and `binary_sensor` here, than I can re-profile with the [test script](https://gist.github.com/jbouwh/10c9ad76a634588bbe891b7e6fd1b6bd), and if everything looks better than we convert the other platforms in followups?

Reply:
done